source/computer/wsl-setup.rst

```diff
@@ -1,18 +1,151 @@
+推荐使用  WSL 安装 Ubuntu 20.04 LTS 或 Ubuntu 20.10。
+
+配置 Linux
+-----------
+
+按照官方文档安装 WSL 之后，就可以根据你的 Linux 发行版进行配置使用了。
+
+如果你安装了 Ubuntu，可以参考 《\ :doc:`/computer/ubuntu-setup` 》进行配置。
+
+X Server
+--------
+
+WLS 本身不支持图形界面，需要在 Windows 中安装 X server
+来接收和显示 Linux 中的图形界面。
+
 .. note::
 
-   尚未完成。
+   X Window System（常称为 X11 或 X）是 UNIX 系统下常用的一种视窗系统，
+   主要由 X Server 和 X Client 两部分组成。其中 X Server 负责接受对图形输出
+   的请求并反馈用户输入，而 X Client 则是使用图形界面的应用程序。
+
+
+Windows 下常见的 X Server 有 `VcXsrv <https://sourceforge.net/projects/vcxsrv/>`__\ 、
+`Xming <http://www.straightrunning.com/XmingNotes/>`__\ 、
+`Xmanager <https://www.xshellcn.com/>`__ 等。
+其中，VcXsrv 是开源免费软件；Xming 和 Xmanager 是收费软件。
+Xming 在 2007 年发布了最后一个免费版本（6.9.0.31）。
+VcXsrv 的使用方式和界面与 Xming 极为相近。
+推荐使用 VcXsrv，本文以此软件为例进行介绍。
+
+1. 下载 `VcXsrv <https://sourceforge.net/projects/vcxsrv/>`__\ ，默认安装即可；
+
+2. 运行 XLaunch，除了在 ``Extra settings`` 界面勾选 ``Disable access control`` 之外，其它选项无需更改；
 
-   WSL 的安装方式可以参考：
+3. 在 Linux 的 ``~/.bashrc`` 文件中添加如下命令后执行 ``source ~/.bashrc``：
+   
+::    
 
-   - 中文指南：https://blog.walterlv.com/post/how-to-install-wsl2.html
-   - 英文指南：https://docs.microsoft.com/en-us/windows/wsl/install-win10
+    # Vcxsrv Display
+    # Windows 每次重启后 WSL2 namesever 的 IP 可能发生变化，如下设置保证能够连接到 XServer
+    export DISPLAY=$(awk '/nameserver / {print $2; exit}' /etc/resolv.conf 2>/dev/null):0
+    export LIBGL_ALWAYS_INDIRECT=1
+
+4、打开图形界面进行测试。
 
 .. note::
 
-   由于维护者目前没有可供安装 Windows 的空闲机器，因而本文短期内无法完成。
-   如果你有 Windows 使用经验，且愿意从头安装 WSL，请到
-   `Issue #65 <https://github.com/seismo-learn/seismology101/issues/65>`__
-   下发表评论。
+   安装并配置好 X Server 之后，切记先运行 XLaunch 再进入 Linux 环境打开图形界面。
+   
+WSL 重要命令
+---------------
+  
+以下命令均在 PowerShell 的管理员模式下运行，并且已安装 WSL2 版本的 Ubuntu20.04LTS，名称为 Ubuntu
+
+::    
+
+    # 查看 wsl 命令帮助
+    $ wsl -h
+    
+    # 列出所有已安装的 Linux 发行版的状态，并显示是 wsl1 还是 wsl2
+    $ wsl -l -v
+    
+    # 停止正在运行的 Linux 发行版，假设其名字为 Ubuntu
+    $ wsl -t Ubuntu
+    
+    # 将 Ubuntu 由 wsl2 更改为 wsl1
+    $ wsl --set-version Ubuntu 1
+    
+    # 将 Ubuntu 由 wsl1 改回 wsl2
+    $ wsl --set-version Ubuntu 2
+    
+    # 进入 Linux 环境
+    $ bash
+    
+    # 退出 Linux 环境
+    $ exit
+    
+    # 将名字为 Ubuntu 的 Linux 发行版导出备份
+    $ mkdir -p D:\WSLBAK  # 创建备份文件夹
+    $ wsl --export Ubuntu D:\WSLBAK\20210117bak.tar   # 备份 wsl 
+    
+    # 将之前备份的 Linux 发行版 Ubuntu 导入还原
+    # 默认的 Linux 发行版都安装在 C 盘，导入还原的时候可以选择其它位置避免 C 盘空间不够
+    $ mkdir -p D:\WSLDIR\Ubuntu20.04  # 创建还原的位置
+    # 还原 wsl 到 D 盘，并命名为 Ubuntu20.04
+    $ wsl --import Ubuntu20.04 D:\WSLDIR\Ubuntu20.04 D:\WSLBAK\20210117bak.tar  
+    
+    # 设置默认发行版
+    # 因为现在有不止一个 Linux 发行版存在，所以需要选择键入 bash 的时候激活哪个
+    $ wsl -s Ubuntu20.04   
+    
+    # 删除某个发行版
+    $ wsl --unregister Ubuntu  # 删除名为 Ubuntu 的发行版
+
+    
+其它
+----
+
+开启 WSL 后 Linux 发行版的默认安装位置在 C 盘，当你安装多个发行版的时候显然会占用 C 盘大量的空间，
+我们可以选择将一个全新安装的 Linux 发行版先导出备份，然后再还原到其它的位置，
+然后将 C 盘上的发行版删除，以保证 C 盘有足够的空间。
+
+WSL1 版本的 Linux 发行版可以和 Windows 系统互相访问文件，而 WSL2 版本的 Linux 文件无法被 Windows 访问。因此，如果想查看
```

Comment:
Do we want to add an example about getting access to files in Linux from Windows and in Windows from Linux?

For example, Disk C is at `/mnt/c` when we are at Linux.
When we are at Windows, how to get access to files in WSL?

Reply:
We can type the follow commands to open the workspace folder in Windows (only in WSL1):
```
$ cd workspace
$ explorer.exe .
```

Reply:
Good~ maybe you can add those in the manual directly?

Reply:
Okay.